diff --git a/pango/pangoft2-private.h b/pango/pangoft2-private.h
index ba16d2de..10c737e5 100644
--- a/pango/pangoft2-private.h
+++ b/pango/pangoft2-private.h
@@ -101,6 +101,10 @@ void  _pango_ft2_font_set_glyph_cache_destroy (PangoFont      *font,
 _PANGO_EXTERN
 GType pango_ft2_renderer_get_type    (void) G_GNUC_CONST;
 
+_PANGO_EXTERN
+void pango_ft2_new_face_substitute ( void *func );
+
 PangoRenderer *_pango_ft2_font_map_get_renderer (PangoFT2FontMap *ft2fontmap);
 
+
 #endif /* __PANGOFT2_PRIVATE_H__ */
diff --git a/pango/pangoft2.c b/pango/pangoft2.c
index b6467981..59d3ead6 100644
--- a/pango/pangoft2.c
+++ b/pango/pangoft2.c
@@ -67,6 +67,8 @@ static void     pango_ft2_font_get_glyph_extents (PangoFont      *font,
 static FT_Face  pango_ft2_font_real_lock_face    (PangoFcFont    *font);
 static void     pango_ft2_font_real_unlock_face  (PangoFcFont    *font);
 
+FT_Error (*FT_New_Face_Func)(FT_Library, const char*, FT_Long, FT_Face*) = &FT_New_Face;
+
 
 PangoFT2Font *
 _pango_ft2_font_new (PangoFT2FontMap *ft2fontmap,
@@ -118,7 +120,7 @@ load_fallback_face (PangoFT2Font *ft2font,
   if (FcPatternGetInteger (matched, FC_INDEX, 0, &id) != FcResultMatch)
     goto bail1;
 
-  error = FT_New_Face (_pango_ft2_font_map_get_library (fcfont->fontmap),
+  error = FT_New_Face_Func (_pango_ft2_font_map_get_library (fcfont->fontmap),
 		       (char *) filename2, id, &ft2font->face);
 
 
@@ -244,7 +246,7 @@ pango_ft2_font_get_face (PangoFont *font)
       if (FcPatternGetInteger (pattern, FC_INDEX, 0, &id) != FcResultMatch)
 	      goto bail0;
 
-      error = FT_New_Face (_pango_ft2_font_map_get_library (fcfont->fontmap),
+      error = FT_New_Face_Func (_pango_ft2_font_map_get_library (fcfont->fontmap),
 			   (char *) filename, id, &ft2font->face);
       if (error != FT_Err_Ok)
 	{
@@ -517,6 +519,16 @@ pango_ft2_get_unknown_glyph (PangoFont *font)
     return PANGO_GLYPH_EMPTY;
 }
 
+void
+pango_ft2_new_face_substitute ( void *func )
+{
+  if (func == NULL) {
+    func = &FT_New_Face;
+  }
+
+  FT_New_Face_Func = func;
+}
+
 void *
 _pango_ft2_font_get_cache_glyph_data (PangoFont *font,
 				     int        glyph_index)
